Hymns/Tags/TagSheetViewModel.swift

```diff
@@ -0,0 +1,95 @@
+import Combine
+import SwiftUI
+import RealmSwift
+import Resolver
+
+class TagSheetViewModel: ObservableObject {
+
+    typealias Title = String
+    @Published var tags: [TagMeta]?
+    @Published var title: String = ""
+
+    let tagStore: TagStore
+    let identifier: HymnIdentifier
+    private let backgroundQueue: DispatchQueue
+    private let mainQueue: DispatchQueue
+    private let repository: HymnsRepository
+    private var disposables = Set<AnyCancellable>()
+
+    init(hymnToDisplay identifier: HymnIdentifier, tagStore: TagStore = Resolver.resolve(), hymnsRepository repository: HymnsRepository = Resolver.resolve(), mainQueue: DispatchQueue = Resolver.resolve(name: "main"), backgroundQueue: DispatchQueue = Resolver.resolve(name: "background")) {
+        self.identifier = identifier
+        self.tagStore = tagStore
+        self.repository = repository
+        self.mainQueue = mainQueue
+        self.backgroundQueue = backgroundQueue
+    }
+
+    func fetchHymn() {
+        repository
+            .getHymn(identifier)
+            .subscribe(on: backgroundQueue)
+            .receive(on: mainQueue)
+            .sink(
+                receiveValue: { [weak self] hymn in
+                    guard let self = self, let hymn = hymn, !hymn.lyrics.isEmpty else {
+                        return
+                    }
+                    let title: Title
+                    if self.identifier.hymnType == .classic {
+                        title = "Hymn \(self.identifier.hymnNumber)"
+                    } else {
+                        title = hymn.title.replacingOccurrences(of: "Hymn: ", with: "")
+                    }
+                    self.title = title
```

Comment:
We're going to need to change this at some point, so could you extract this into a common function or something?

Maybe you can make the `title` field in UiHymn a computed property and stick this in there. That way both DisplayHymnViewModel and this view model can both just call it

Reply:
Nice! done in recent commit